init/init.c

```diff
@@ -61,55 +65,170 @@ static void set_rlimits(const char *rlimits)
+                perror("mount(/dev/vdb)");
+
+                goto rmdir_teeconfig;
+        }
+
+        fd = open("/teeconfig/snp-config-attest.json", O_RDONLY);
+        if (fd < 0) {
+                perror("open(snp-config-attest.json)");
+
+                goto umount_teeconfig;
+        }
+
+        if (read(fd, (void *) tc_json, 1024) < 0) {
+                perror("read(snp-config-attest.json)");
+                close(fd);
+
+                goto umount_teeconfig;
+        }
+        close(fd);
+
+        /*
+         * Unmount and remove the mounted directory.
+         */
+        if (umount("/teeconfig") < 0)
+                printf("Unable to unmount /teeconfig");
+
+        teeconfig = opendir("/teeconfig");
+        if (teeconfig == NULL) {
+                printf("Unable to open /teeconfig directory\n");
+
+                goto umount_teeconfig;
+        }
+
+        while ((tc_de = readdir(teeconfig)) != NULL) {
+                stat(tc_de->d_name, &tc_stat);
+                if (!strcmp(".", tc_de->d_name) || !strcmp("..", tc_de->d_name))
+                        continue;
+
+                sprintf(full_path, "/teeconfig/%s", tc_de->d_name);
+                if (remove(full_path) < 0)
+                        printf("Unable to remove file %s\n", full_path);
+        }
+        if (rmdir("/teeconfig") < 0)
+                printf("Unable to remove directory /teeconfig\n");
+
+        /*
+         * Parse the TEE config's workload_id and attestation_url field.
+         */
+        jsmn_init(&parser);
+
+        tokens = (jsmntok_t *) malloc(sizeof(jsmntok_t) * MAX_TOKENS);\
+        if (tokens == NULL) {
+                perror("malloc(jsmntok_t)");
+
+                goto umount_teeconfig;
+        }
+
+        num_tokens = jsmn_parse(&parser, tc_json, strlen(tc_json), tokens,
+                MAX_TOKENS);
+        if (num_tokens < 0) {
+                printf("Unable to allocate JSON tokens\n");
+
+                goto umount_teeconfig;
+        } else if (num_tokens < 1 || tokens[0].type != JSMN_OBJECT) {
+                printf("Unable to find object in TEE configuration file\n");
+
+                goto umount_teeconfig;
+        }
+
+        wid_found = url_found = 0;
+
+        for (int i = 1; i < num_tokens - 1; ++i) {
+                tok_start = tc_json + tokens[i + 1].start;
+                tok_end = tc_json + tokens[i + 1].end;
+                tok_size = tok_end - tok_start;
+                if (!jsoneq(tc_json, &tokens[i], "workload_id")) {
+                        strncpy(wid, tok_start, tok_size);
+                        wid_found = 1;
+                } else if (!jsoneq(tc_json, &tokens[i], "attestation_url")) {
+                        strncpy(url, tok_start, tok_size);
+                        url_found = 1;
+                }
+        }
 
-	if (mkdir("/sfs", 0755) < 0 && errno != EEXIST) {
-		perror("mkdir(/sfs)");
-		return NULL;
-	}
-
-	if (mount("securityfs", "/sfs", "securityfs",
```

Comment:
We still need this code for SEV/SEV-ES. After opening `krun-sev.json`, we need to check the "tee" field and either use this code or, for the SNP case, use the new one.

Reply:
Fixed in [dcd5828](https://github.com/containers/libkrun/pull/108/commits/dcd5828f89215578f69b47541bd146490de6d037)